packages/back-end/src/controllers/informationSchemas.ts

```diff
@@ -224,8 +224,8 @@ export async function putInformationSchema(
   }
 
   req.checkPermissions(
-    "editDatasourceSettings",
-    datasource?.projects?.length ? datasource.projects : ""
+    "runQueries",
+    datasource?.projects?.length ? datasource.projects : []
```

Comment:
This is an unrelated bug - that no one has ran into yet, but.... if a user has permission to runQueries, they can open the `EditSqlModal` which renders the `SchemaBrowser`. 

And if the `informationSchema` is over 30 days old, we try to refresh it in the background, and we were checking for `editDatasourceSettings`, which is a permission that the user must either have globally (if the resource doesn't have any projects).

So if the user tried to edit a query on a datasource that was in "All Projects" and the user had a global role that doesn't permit editing a data source, the whole app would crash.

Reply:
Agreed with this change.